### PR TITLE
New package: materia-gtk-theme-20190831

### PR DESCRIPTION
**Pull request by lugubris**
fixes https://github.com/void-linux/void-packages/issues/14246

### CODE REVIEW

**Comment by lugubris** [comment]
> => ERROR: materia-gtk-theme-20190831_1: cannot access wrksrc directory [/builddir/materia-gtk-theme]

Why is the build failing?
adding:
```
mkdir ${wrksrc}
```
doesn't help

**Comment by lugubris** [comment]
What will be better `wrksrc` equals `materia-gtk-theme` or `materia-gtk-theme-v${version}`?

hostmake currently depends on git, is there a better way?

**Comment by lugubris** [comment]
I need some help with this template.
```
=> ERROR: materia-gtk-theme-20190831_1: cannot find v20190831.tar.gz, use 'xbps-src fetch' first.

The command "if [ -z "$XLINT" ]; then docker exec -t void hostrepo/common/travis/build.sh $BOOTSTRAP $ARCH; fi" exited with 1.
```

**Comment by lugubris** [comment]
I need help with this template.
```
Please run as root.

=> ERROR: materia-gtk-theme-20190831_1: do_install: './install.sh' exited with 1

=> ERROR:   in do_install() at srcpkgs/materia-gtk-theme/template:19
```

**Comment by zdtcd** [comment]
Here is a working template:
```sh
# Template file for 'materia-gtk-theme'
pkgname=materia-gtk-theme
version=20190831
revision=1
archs=noarch
wrksrc="materia-theme-${version}"
depends="gtk+3 gnome-themes-extra"
short_desc="Material Design theme for GNOME/GTK based desktop environments"
maintainer="Lugubris <lugubris@disroot.org>"
license="GPL-2.0-only"
homepage=https://github.com/nana-4/materia-theme
distfiles="https://github.com/nana-4/materia-theme/archive/v${version}.tar.gz"
checksum=7ab5423ecaef50d5e9cf94038f9a35a1bcb87e7b218edff316b4f92d4b7b8b67

do_install() {
	vmkdir usr/share/themes
	./install.sh -d "${DESTDIR}/usr/share/themes"
}
```

**Comment by lugubris** [comment]
Danh Doan @ 2019-09-08 01:42 UTC:

> Here is a working template:
> ```sh
> # Template file for 'materia-gtk-theme'
> pkgname=materia-gtk-theme
> version=20190831
> revision=1
> archs=noarch
> wrksrc="materia-theme-${version}"
> depends="gtk+3 gnome-themes-extra"
> short_desc="Material Design theme for GNOME/GTK based desktop environments"
> maintainer="Lugubris <lugubris@disroot.org>"
> license="GPL-2.0-only"
> homepage=https://github.com/nana-4/materia-theme
> distfiles="https://github.com/nana-4/materia-theme/archive/v${version}.tar.gz"
> checksum=7ab5423ecaef50d5e9cf94038f9a35a1bcb87e7b218edff316b4f92d4b7b8b67
>
> do_install() {
> 	vmkdir usr/share/themes
> 	./install.sh -d "${DESTDIR}/usr/share/themes"
> }
> ```

That didn't work, I had some problems with the tar command, current
template works.

relevant: https://www.xkcd.com/1168/

foo.tar has a directory bar so the structure is:

foo.tar
-bar
--file1
--subdir1
--file2
..

When I extract foo.tar it creates a directory bar, like this:

wrksrc
-foo.tar
-bar
--file1
..

I want it to extract all contents of bar to wrksrc, like:

wrksrc
-foo.tar
-file1
-subdir1
-file2
..

Currently I cd into bar and then run ./install which works so I guess
its fine.
--
                                Lugubris

**Comment by zdtcd** [comment]
On 2019-09-09 00:12:50 -0700, Lugubris wrote:
> 
> 
> That didn't work, I had some problems with the tar command, current
> template works.

Have you tried to build with my suggested template?

I tested on my local machine, and it works fine.

> I want it to extract all contents of bar to wrksrc, like:

Why do you need to extract everything into $wrksrc?

> Currently I cd into bar and then run ./install which works so I guess

With my suggested template, you don't need to `cd` anywhere.

-- 
Danh

File: srcpkgs/materia-gtk-theme/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'materia-gtk-theme'
+pkgname=materia-gtk-theme
+version=20190831
+revision=1
+archs=noarch
+wrksrc="materia-gtk-theme-v${version}"
```
Comment (zdtcd) [inline srcpkgs/materia-gtk-theme/template]: `s/v//`

With this change, you won't need to `untar` in `do_install`

File: srcpkgs/materia-gtk-theme/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+depends="gtk+3 gnome-themes-extra"
+short_desc="Material Design theme for GNOME/GTK based desktop environments"
+maintainer="Lugubris <lugubris@disroot.org>"
+license="GPL-2.0-only"
+homepage=https://github.com/nana-4/materia-theme
+distfiles="https://github.com/nana-4/materia-theme/archive/v${version}.tar.gz"
+checksum=7ab5423ecaef50d5e9cf94038f9a35a1bcb87e7b218edff316b4f92d4b7b8b67
+
+do_install() {
+	tar -C ${wrksrc} -xzf v${version}.tar.gz
```
Comment (zdtcd) [inline srcpkgs/materia-gtk-theme/template]: With my suggested change in `wrksrc`, we don't need this `tar && cd`

File: srcpkgs/materia-gtk-theme/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'materia-gtk-theme'
+pkgname=materia-gtk-theme
+version=20190831
+revision=1
+archs=noarch
+wrksrc="materia-gtk-theme-v${version}"
+build_style=fetch
```
Comment (zdtcd) [inline srcpkgs/materia-gtk-theme/template]: `7d`

**Comment by lugubris** [comment]
Danh Doan @ 2019-09-09 07:23 UTC:

> On 2019-09-09 00:12:50 -0700, Lugubris wrote:
>>
>>
>> That didn't work, I had some problems with the tar command, current
>> template works.
>
> Have you tried to build with my suggested template?

Your template didn't work on my local machine because I didn't remove -v
part from wrksrc (the change you suggested). My bad, should work now.

--
                                Lugubris

**Comment by Chocimier** [comment]
Usually we do not accept packages that can be put into $HOME, is this any different?

**Comment by hfel** [comment]
Was a final decision reached regarding this package?

**Pull request closed without merging**